### PR TITLE
Adding Joiner markdown

### PR DESCRIPTION
**Pull request by CuriousVini**
Issue: https://issues.cask.co/browse/HYDRA-189
Build: http://builds.cask.co/browse/HYP-BAD47-3

### CODE REVIEW

File: core-plugins/docs/Joiner-batchjoiner.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# Joiner
+
+
+Description
+-----------
+Joins records from one or more sources based on join keys. Supports `inner` and `outer` joins, selection and renaming of output fields.
```
Comment (sreevatsanraman) [inline core-plugins/docs/Joiner-batchjoiner.md]: Is this limited to just sources? Should we just say input?

File: core-plugins/docs/Joiner-batchjoiner.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
+# Joiner
+
+
+Description
+-----------
+Joins records from one or more sources based on join keys. Supports `inner` and `outer` joins, selection and renaming of output fields.  
+
+Use Case
+--------
+The transform is used when you want to join records from one or more sources similar to joins in SQL.
```
Comment (sreevatsanraman) [inline core-plugins/docs/Joiner-batchjoiner.md]: The transform is used when you want to combine fields from one or more input, similar to the joins in SQL.

File: core-plugins/docs/Joiner-batchjoiner.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+Properties
+----------
+**joinKeys:** List of join keys to perform join operation. The list is separated by `&`.
```
Comment (sreevatsanraman) [inline core-plugins/docs/Joiner-batchjoiner.md]: List of keys to perform the join operation.

File: core-plugins/docs/Joiner-batchjoiner.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+Properties
+----------
+**joinKeys:** List of join keys to perform join operation. The list is separated by `&`. 
+Join key from each input stage will be prefixed with `<stageName>.` and the relation among join keys from different inputs is represented by `=`. 
+For example: customers.customer_id=items.c_id&customers.customer_name=items.c_name means the join key is a composite key
+of customer id and customer name from customers and items input stages and join will be performed on equality 
+of the join keys. This transform only supports equality for joins.
+
+**selectedFields:** Comma-separated list of fields to be selected and renamed in join output from each input stage. 
+Each selected input field name needs to be present in the output must be prefixed with '<input_stage_name>'.
```
Comment (sreevatsanraman) [inline core-plugins/docs/Joiner-batchjoiner.md]: Each selected field that should be present in the output ...

**Comment by sreevatsanraman** [comment]
Minor comments. Fix and merge.

**Comment by CuriousVini** [comment]
Squashing and merging.

**Pull request merged**